keps/sig-network/20190125-ingress-api-group.md

```diff
@@ -233,9 +234,16 @@ Note: default value are permitted between API versions
 
 [api-conv-versions]: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#defaulting
 
+##### Defaults
+
+For a smooth upgrade process, the `PathType` field will be added to the `v1beta1`
+Ingress API. This will be an optional field in `v1beta1` defaulting to a value
+of `ImplementationSpecific`. In `v1`, the field will be a required field (i.e. 
```

Comment:
Why make it required in v1, rather than default to "Exact" ?  It's more obvious, but somewhat draconian.  Not a hard requirement, I just want to see the rationale holds water.

Reply:
I think that would make for a poor user experience. It would make a significant change to ingress behavior, and that change would be easy to miss with a default value getting set on a new field. As an end user, you'd have to be very up to date with Kubernetes development to understand why your new ingresses weren't working as expected. I think the most compelling options are either to have the default match already established behavior, or not have a default so input is required and no one can miss this change. This proposes both of those approaches (v1beta1 default matching already established behavior, v1 requiring user input to improve visibility and adoption of new path types).

Reply:
OK, I can buy that